mne/inverse_sparse/mxne_inverse.py

```diff
@@ -133,7 +140,10 @@ def _make_sparse_stc(X, active_set, forward, tmin, tstep,
             n_points_so_far = this_n_points_so_far
             vertices.append(this_vertno)
 
-        stc = SourceEstimate(X, vertices=vertices, tmin=tmin, tstep=tstep)
+        if pick_ori == 'vector':
+            stc = VectorSourceEstimate(X, vertices, tmin, tstep)
+        else:
+            stc = SourceEstimate(X, vertices, tmin, tstep)
```

Comment:
@larsoner would you mind looking at how to factorize this code with the few lines are the end of `apply_inverse` 
 functions. That would avoid bugs due our variety of source space types.

Reply:
Did you want to address this within this PR? otherwise I think I addressed all the issues you raised.

Reply:
I'm happy to do it either way, maybe merge this then I'll refactor/unify?